index.ts

```diff
@@ -106,6 +107,12 @@ if (!oauthEnabled) {
         .join(" ")
         .trim();
 
+      req.session.uid = process.env.OAUTH_USER_ID_PATH.split("|")
+        .map(path => getPropertyByPath(userData, path))
+        .join("_")
+        .replace(/[^_a-zA-Z0-9-]/g, '_')
```

Comment:
I am not all too happy with this solution, because it could lead to collisions in edge cases where the difference between IDs consists of illegal characters.

Reply:
Me neither :D and for the same reasons.

How about taking the uid as is but not using it on the client-side but for that instead use an ID from `generateId()`? This would require extending `interface Connection` and some minor changes of how it's used?!

Reply:
Since we heavily rely on the id when communicating with the clients I would see this as error prone because you can easily forget mapping the values. I already looked into the `crypto` library and thought about using a hex-representation of the ids md5-hash (or sha-256 if we want to be overly cautious).